numpy/lib/nanfunctions.py

```diff
@@ -508,7 +515,7 @@ def nansum(a, axis=None, dtype=None, out=None, keepdims=np._NoValue):
 
     Numpy integer arithmetic is modular. If the size of a sum exceeds the
```

Comment:
This entire note seems out of place for nan functions. There isn't an integer NaN, so even though the functions work on integer arrays, the warning is misleading. If anything, these functions should advice using the normal `np.sum` function for integers, rather than warning about integer overflow.